### DISCUSSION TITLE
Filter search

### DISCUSSION THREAD

**Discussion by seabird2fr** [symfony/symfony#45178] [Q&A]
Hello,

I just noticed something strange.
I made a product search filter by category
I have my class below: 

```
class Search
{

    private $categories;

        public function getCategories()
        {
            return $this->categories;
        }
    
        public function setCategories($categories): self
        {
            $this->categories = $categories;
            return $this;
        }
}

```

and the form type
```
class SearchType extends AbstractType
{
    public function buildForm(FormBuilderInterface $builder, array $options): void
    {
        $builder
            ->add('categories',EntityType::class,[
                'class'=>Category::class,
                'choice_label'=>'name',
                'required'=>false,
                'multiple'=>true,
                'expanded'=>true
 
            ])
        ->add('submit',SubmitType::class,[
            'label'=>'Filtrer',
            'attr'=>[
                'class'=>'btn col-12 btn-primary'
            ]
 
            ]);
    }
 
    public function configureOptions(OptionsResolver $resolver): void
    {
        $resolver->setDefaults([
            // Configure your form options here
            'data_class' => Search::class,
            'method'=>'GET'
        ]);
    }
}
```

I did not put the controller
The strange phenomenon is happening on the twig side.
If I put the code

```
{{form_start(form)}}
 
 {{form_end(form)}}
```

it's OK 
But if I put
```
{{form(form)}}
```

I have an error: **Object of class App\Entity\Search could not be converted to string**
And if I put a function __toString a field search appears 

```
<input type="search" id="search" name="search" required="required" class="form-control">

```

Obviously, it's because of the name that I used **Search.php** and **SearchType.php**. If I change the names everything goes back to normal. 

Eric

**Comment by stof** [comment]
This is because the default block prefix is generated based on the short class name only in AbstractType. So your `App\Form\Type\SearchType` class gets `search` as its block prefix, which is already used by `Symfony\Component\Form\Extension\Core\Type\SearchType` and the form theme renders `search_widget` as a `<input type=search>`.

The solution in your case is to define `getBlockPrefix` explicitly in your own SearchType to return a value that does not conflict with the core SearchType (and for which the form theme will probably not provide a custom rendering, as the rendering inherited from FormType will probably be enough). For instance, you could use `app_search`
